packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx

```diff
@@ -443,7 +443,7 @@ export const MultiSectionDigitalClock = React.forwardRef(function MultiSectionDi
           items={viewTimeOptions[timeView].items}
           onChange={viewTimeOptions[timeView].onChange}
           active={view === timeView}
-          autoFocus={autoFocus ?? focusedView === timeView}
+          autoFocus={autoFocus || focusedView === timeView}
```

Comment:
Having an explicit `autoFocus` value of `false` prevented the focus from jumping to this component on `DesktopDateTimePicker`. 🙈